File: Samples-Java/Samples/src/main/java/fragments/autocomplete/AutoCompleteTokenLayoutsFragment.java
package fragments.autocomplete;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.telerik.android.sdk.R;
import com.telerik.widget.autocomplete.AutoCompleteAdapter;
import com.telerik.widget.autocomplete.CompletionMode;
import com.telerik.widget.autocomplete.DisplayMode;
import com.telerik.widget.autocomplete.LayoutMode;
import com.telerik.widget.autocomplete.RadAutoCompleteTextView;
import com.telerik.widget.autocomplete.SuggestMode;
import com.telerik.widget.autocomplete.TokenModel;

import java.util.ArrayList;

import activities.ExampleFragment;

public class AutoCompleteTokenLayoutsFragment extends JsonDataLoadFragment implements ExampleFragment {

    private String[] data = new String[]{"Australia", "Albania","Bulgaria","Belgium","Cyprus","Italy","Japan",
            "Denmark","Finland","France","Germany","Greece","Hungary","Ireland",
            "Latvia","Luxembourg","Macedonia","Moldova","Monaco","Netherlands","Norway",
            "Poland","Romania","Russia","Sweden","Slovenia","Slovakia","Turkey","Ukraine",
            "Vatican City"};
    private RadAutoCompleteTextView autocomplete = null;

    public String title() {
        return "Token Layouts";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.autocomplete_token_layouts, container, false);

        autocomplete = (RadAutoCompleteTextView) rootView.findViewById(R.id.autocmp);
        autocomplete.setSuggestMode(SuggestMode.SUGGEST);
        autocomplete.setDisplayMode(DisplayMode.TOKENS);
        autocomplete.setTokensLayoutMode(LayoutMode.HORIZONTAL);

        final AutoCompleteAdapter adapter = new AutoCompleteAdapter(this.getContext(),this.getTokenModelObjects(), R.layout.suggestion_item_layout);
        adapter.setCompletionMode(CompletionMode.STARTS_WITH);
        autocomplete.setAdapter(adapter);

        this.setButtonActions(rootView);

        return rootView;
    }

    private ArrayList<TokenModel> getTokenModelObjects() {
        ArrayList<TokenModel> feedData = new ArrayList<TokenModel>();
        for(int i = 0; i < this.data.length; i++){
            TokenModel token = new TokenModel(this.data[i], null, null);
            feedData.add(token);
        }

        return feedData;
    }

    private void setButtonActions(View rootView){
        Button horizontalBtn = (Button)rootView.findViewById(R.id.horizontal_layout_btn);
        Button wrapBtn = (Button)rootView.findViewById(R.id.wrap_layout_btn);

        horizontalBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setTokensLayoutMode(LayoutMode.HORIZONTAL);
                autocomplete.resetAutocomplete();
            }
        });

        wrapBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setTokensLayoutMode(LayoutMode.WRAP);
                autocomplete.resetAutocomplete();
            }
        });
    }
}

File: Samples-Xamarin/Fragments/Autocomplete/AutoCompleteTokenLayoutsFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Autocomplete;

namespace Samples
{
	public class AutoCompleteTokenLayoutsFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		private RadAutoCompleteTextView autocomplete;
		private AutoCompleteAdapter adapter;
		private List<String> data;

		public AutoCompleteTokenLayoutsFragment() : base()
		{
			this.data = new List<string>() { "Australia", "Albania","Austria", "Argentina", "Maldives","Bulgaria","Belgium","Cyprus","Italy","Japan",
										"Denmark","Finland","France","Germany","Greece","Hungary","Ireland",
										"Latvia","Luxembourg","Macedonia","Moldova","Monaco","Netherlands","Norway",
										"Poland","Romania","Russia","Sweden","Slovenia","Slovakia","Turkey","Ukraine",
										"Vatican City", "Chad", "China", "Chile" };
		}

		public string Title()
		{
			return "Tokens Layout";
		}

		public override Android.Views.View OnCreateView(Android.Views.LayoutInflater inflater, Android.Views.ViewGroup container, Android.OS.Bundle savedInstanceState)
		{
			View rootView = inflater.Inflate(Resource.Layout.fragment_autocomplete_token_layouts, container, false);
			this.autocomplete = (RadAutoCompleteTextView)rootView.FindViewById(Resource.Id.autocmp);

			this.autocomplete.SuggestMode = SuggestMode.Suggest;
			// >> autocomplete-display-mode-xamarin
			this.autocomplete.DisplayMode = DisplayMode.Tokens;
			// << autocomplete-display-mode-xamarin

			// >> autocomplete-layout-mode-xamarin
			this.autocomplete.TokensLayoutMode = LayoutMode.Horizontal;
			// << autocomplete-layout-mode-xamarin

			this.adapter = new AutoCompleteAdapter(this.Context, this.GetTokenObjects(), Java.Lang.Integer.ValueOf(Resource.Layout.suggestion_item_layout));

			// >> autocomplete-completion-mode-xamarin
			this.adapter.CompletionMode = CompletionMode.StartsWith;
			// << autocomplete-completion-mode-xamarin
			this.autocomplete.Adapter = this.adapter;

			Display display = this.Activity.WindowManager.DefaultDisplay;
			int height = display.Height;
			this.autocomplete.SuggestionViewHeight = height / 4;

			this.SetButtonActions(rootView);

			return rootView;
		}

		private List<TokenModel> GetTokenObjects()
		{
			List<TokenModel> feedData = new List<TokenModel>();
			for (int i = 0; i < this.data.Count; i++)
			{
				TokenModel token = new TokenModel(this.data[i], null);
				feedData.Add(token);
			}

			return feedData;
		}

		private void SetButtonActions(View rootView)
		{
			Button btnHorizontal = (Button)rootView.FindViewById(Resource.Id.horizontal_layout_btn);
			btnHorizontal.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.TokensLayoutMode = LayoutMode.Horizontal;
				this.autocomplete.ResetAutocomplete();
			};

			Button btnWrap = (Button)rootView.FindViewById(Resource.Id.wrap_layout_btn);
			btnWrap.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.TokensLayoutMode = LayoutMode.Wrap;
				this.autocomplete.ResetAutocomplete();
			};
		}
	}
}
